src/WebJobs.Extensions.DurableTask/DurableOrchestrationClient.cs

```diff
@@ -172,5 +165,190 @@ private static JToken ParseToJToken(string value)
+                    {
+                        historyArray.RemoveAt(indexValue - counter);
+                        counter++;
+                    }
+                }
+            }
+
+            return new DurableOrchestrationStatus
+            {
+                Name = orchestrationState.Name,
+                InstanceId = orchestrationState.OrchestrationInstance.InstanceId,
+                CreatedTime = orchestrationState.CreatedTime,
+                LastUpdatedTime = orchestrationState.LastUpdatedTime,
+                RuntimeStatus = (OrchestrationRuntimeStatus)orchestrationState.OrchestrationStatus,
+                Input = ParseToJToken(orchestrationState.Input),
+                Output = ParseToJToken(orchestrationState.Output),
+                History = historyArray,
+            };
+        }
+
+        private static void UpdateProperties(ref JToken jsonToken, List<string> propertyNames, bool hideInput = false, bool hideResult = false)
+        {
+            if (jsonToken == null)
+            {
+                return;
+            }
+
+            if ((hideInput || hideResult) && propertyNames == null)
+            {
+                propertyNames = new List<string>();
+            }
+
+            if (hideInput)
+            {
+                propertyNames.Add("Input");
+            }
+
+            if (hideResult)
+            {
+                propertyNames.Add("Result");
+            }
+
+            foreach (var propertyName in propertyNames)
+            {
+                UpdateProperties(ref jsonToken, propertyName);
+            }
+        }
+
+        private static void UpdateProperties(ref JToken jsonToken, string properyName, bool hideInput = false, bool hideResult = false)
+        {
+            if (jsonToken == null)
+            {
+                return;
+            }
+
+            if (hideInput)
+            {
+                jsonToken["Input"]?.Parent?.Remove();
+            }
+
+            if (hideResult)
+            {
+                jsonToken["Result"]?.Parent?.Remove();
+            }
+
+            if (!string.IsNullOrEmpty(properyName))
+            {
+                jsonToken[properyName]?.Parent?.Remove();
+            }
+        }
+
+        private static void TrackNameAndScheduledTime(JToken historyItem, EventType eventType, int index, ref Dictionary<string, EventIndexDateMapping> eventMapper)
+        {
+            eventMapper.Add($"{eventType}_{historyItem["EventId"]}", new EventIndexDateMapping { Index = index, Name = historyItem["Name"].ToString(), Input = historyItem["Input"]?.ToString(), Date = (DateTime)historyItem["Timestamp"] });
+        }
+
+        private static void AddScheduledEventDataAndAggreagate(IReadOnlyDictionary<string, EventIndexDateMapping> eventMapper, string prefix, JToken historyItem, ref List<int> indexList)
+        {
+            var taskScheduledData = eventMapper[$"{prefix}_{historyItem["TaskScheduledId"]}"];
```

Comment:
I wonder if we should have a null-check here in case the event data comes in out of order for some reason.

Reply:
Yes, great point. I think it is a case we need to check for. I will add it.

Thank you!